buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -103,16 +103,76 @@ class BuildPlugin implements Plugin<Project> {
 
     /** Makes dependencies non-transitive by default */
     static void configureConfigurations(Project project) {
+        // fail on any conflicting dependency versions
+        project.configurations.all({ Configuration configuration ->
```

Comment:
The `_transitive_` hack is probably worth of a 1-2 paragraph treatise somewhere above.